src/ParallelAlgorithms/EventsAndTriggers/CMakeLists.txt

```diff
@@ -3,7 +3,13 @@
 
 set(LIBRARY EventsAndTriggers)
 
-add_spectre_library(${LIBRARY} INTERFACE)
+add_spectre_library(${LIBRARY})
+
+spectre_target_sources(
+  ${LIBRARY}
+  PRIVATE
+  LogicalTriggers.cpp
```

Comment:
Since this file includes `LogicalTriggers.hpp` you'll need to link all libs that the header depends on as `PUBLIC` now, instead of `INTERFACE`. Those libs are now needed to compile the `EventsAndTriggers` lib. They are: `Options`, `Parallel`, `Utilities`